runtime/drivers/clickhouse/model_executor_s3_self.go

```diff
@@ -26,6 +26,21 @@ func (p *inputProps) Validate() error {
 	return nil
 }
 
+type outputProperties struct {
+	ModelOutputProperties `mapstructure:",squash"`
+}
+
+func (p *outputProperties) Validate() error {
+	// set output props with defaults
+	if p.Materialize == nil {
+		materialize := true
+		p.Materialize = &materialize
+	} else if !*p.Materialize {
+		return fmt.Errorf("s3 to clickhouse executor only supports materialized models. Set `materialize` to true")
+	}
+	return nil
+}
```

Comment:
So now these changes aren't needed, right?

Reply:
This is still needed.
In the model_executor it doesn't have context of s3. The input and output connector are both clickhouse in that case and it is not necessary that incremental and split are true so we need to make sure that materialize is true by default.
The `model_executor_s3_self` just calls `model_executor_self` with a SQL `SELECT * FROM S3(...`

Reply:
Okay I see – it's not very clear from the context, so it would be nice with a comment and maybe to combine it into one section of code, especially given the output properties aren't accessed or used for other purposes here. What do you think of my suggestion below?